src/test/java/com/imgix/test/TestAll.java

```diff
@@ -100,6 +100,31 @@ public void testHelperBuildNestedPath() {
               "example/chester.png",
               "http://securejackangers.imgix.net/example/chester.png?w=500"
             },
+            {
+              "Relative Path With Unencoded Space",
+              "example/chester 1.png",
+              "http://securejackangers.imgix.net/example/chester%201.png?w=500"
+            },
+            {
+              "Relative Path With Unencoded Plus",
+              "example/chester" + "+1" + ".png",
+              "http://securejackangers.imgix.net/example/chester%2B1.png?w=500"
+            },
+            {
+              "Relative Path With Unencoded Colon",
+              "example/chester" + ":1" + ".png",
+              "http://securejackangers.imgix.net/example/chester%3A1.png?w=500"
+            },
+            {
+              "Relative Path With Unencoded Question Mark",
+              "example/chester" + "?1" + ".png",
+              "http://securejackangers.imgix.net/example/chester%3F1.png?w=500"
+            },
+            {
+              "Relative Path With Unencoded Hash",
+              "example/chester" + "#1" + ".png",
+              "http://securejackangers.imgix.net/example/chester%231.png?w=500"
+            },
```

Comment:
Test that relative paths, ie `somePath/foo/bar`, are encoding ` +:?$` characters.

Reply:
Use these paths as well.

```
path = "/&$+,:;=?@#.jpg"

path = "/ <>[]{}|\\^%.jpg"

path = "/ساندویچ.jpg"
```

We don't (yet) have a unified testing suite that will test that all encodings resolve, but we should coordinate to make sure they do. When you encode with these paths lmk and we'll attach them to @sherwinski domain/source to ensure they resolve.

I've come to care a lot less about uniform percent encoding output and more about "does this resolve or not?" Haha

Reply:
Adding this to the stacked PR instead. It addresses those paths better.